pytorch_lightning/trainer/deprecated_api.py

```diff
@@ -0,0 +1,135 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pytorch_lightning.utilities import rank_zero_warn, DistributedType, DeviceType
+
+
+class DeprecatedDistDeviceAttributes:
+
+    _distrib_type: DistributedType
+    _device_type: DeviceType
+    num_gpus: int
+
+    @property
+    def on_cpu(self) -> bool:
+        # rank_zero_warn("Internal: `on_cpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
+        return self._device_type and self._device_type == DeviceType.CPU
+
+    @on_cpu.setter
+    def on_cpu(self, val: bool) -> None:
+        rank_zero_warn("Internal: `on_cpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
+        if val:
+            self._device_type = DeviceType.CPU
+
+    @property
+    def on_tpu(self) -> bool:
+        # rank_zero_warn("Internal: `on_tpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
+        return self._device_type and self._device_type == DeviceType.TPU
+
+    @on_tpu.setter
+    def on_tpu(self, val: bool) -> None:
+        rank_zero_warn("Internal: `on_tpu` is deprecated in v1.1 and will be removed in v1.2.", DeprecationWarning)
+        # todo add logic that it cannot be set if TPU is missing
+        if val:
+            self._device_type = DeviceType.TPU
```

Comment:
If you add these, the user will see many of the warnings because internally the setter is used.
Is the deprecation really necessary? The user is not supposed to interact with these attributes in the first place, right?

Reply:
ok, I ll drop the warning for now and add it after we replace it in code...